pydantic_settings/sources.py

```diff
@@ -121,7 +121,7 @@ def get_field_value(self, field: FieldInfo, field_name: str) -> tuple[Any, str,
         return None, '', False
 
     def __call__(self) -> dict[str, Any]:
-        return self.init_kwargs
+        return TypeAdapter(dict[str, Any]).dump_python(self.init_kwargs)
```

Comment:
It's failing on python 3.8
```suggestion
        return TypeAdapter(Dict[str, Any]).dump_python(self.init_kwargs)
```